src/transformers/hf_argparser.py

```diff
@@ -66,7 +67,10 @@ def __init__(self, dataclass_types: Union[DataClassType, Iterable[DataClassType]
         super().__init__(**kwargs)
         if dataclasses.is_dataclass(dataclass_types):
             dataclass_types = [dataclass_types]
-        self.dataclass_types = dataclass_types
+        if isinstance(dataclass_types, collections.Sequence):
+            self.dataclass_types = dataclass_types
+        else:
+            self.dataclass_types = list(dataclass_types)
```

Comment:
Is there a downside in always doing the `list`?

Reply:
Extra memory is created if it's already a sequence. Though it should be tiny. Should I change it?

Reply:
I think it would make the code easier to read, and I don't think the list of arguments is going to be that big in memory :-)